test/integration-tests/Seaport.test.ts

```diff
@@ -141,4 +157,187 @@ describe('Seaport', () => {
+    })
+
+    it('revertable order returns NFT to user if reverts', async () => {
+      let { advancedOrder, criteriaResolvers } = getAdvancedOrderParams(seaportOrders[2])
+      const id = criteriaResolvers[0].identifier
+      // transfer nft to alice as tubbyCatOwner
+      const prevTubbyCatOwner = await tubbyCats.connect(alice).ownerOf(id)
+      await tubbyCats
+        .connect(await ethers.getImpersonatedSigner(prevTubbyCatOwner))
+        .transferFrom(prevTubbyCatOwner, alice.address, id)
+      tubbyCats = tubbyCats.connect(alice)
+      const ownerBefore = await tubbyCats.ownerOf(id)
+      expect(ownerBefore).to.eq(alice.address)
+
+      // Signature is invalid
+      advancedOrder.signature = '0xdeadbeef'
+
+      const calldata = seaportInterface.encodeFunctionData('fulfillAdvancedOrder', [
+        advancedOrder,
+        criteriaResolvers,
+        OPENSEA_CONDUIT_KEY,
+        ADDRESS_ZERO, // 0 addr so router custody
+      ])
+
+      planner.addCommand(
+        CommandType.SEAPORT_SELL_721,
+        [calldata, tubbyCats.address, OPENSEA_CONDUIT, id, alice.address],
+        true
+      )
+      planner.addCommand(CommandType.SWEEP, [WETH.address, MSG_SENDER, 0])
+      const { commands, inputs } = planner
+
+      // TODO: replace with permit2 transfer
+      await tubbyCats.transferFrom(alice.address, router.address, id)
+
+      const wethBefore = await weth.connect(alice).balanceOf(alice.address)
+      expect(wethBefore).to.eq(0)
+      const ethBefore = await ethers.provider.getBalance(alice.address)
+      const receipt = await (await router['execute(bytes,bytes[],uint256)'](commands, inputs, DEADLINE, {})).wait()
+
+      const ownerAfter = await tubbyCats.ownerOf(id)
+      const wethAfter = await weth.connect(alice).balanceOf(alice.address)
+      const ethAfter = await ethers.provider.getBalance(alice.address)
+      const gasSpent = getTxGasSpent(receipt)
+      const ethDelta = ethBefore.sub(ethAfter)
+
+      // owner never changed and is alice still
+      expect(ownerBefore).to.eq(ownerAfter)
+      expect(wethAfter.eq(wethBefore)).to.be.true
+      expect(ethDelta).to.eq(gasSpent)
+    })
+
+    it('reverts if order does not go through', async () => {
+      let { advancedOrder, criteriaResolvers } = getAdvancedOrderParams(seaportOrders[2])
+      const id = criteriaResolvers[0].identifier
+      // transfer nft to alice as tubbyCatOwner
+      const prevTubbyCatOwner = await tubbyCats.connect(alice).ownerOf(id)
+      await tubbyCats
+        .connect(await ethers.getImpersonatedSigner(prevTubbyCatOwner))
+        .transferFrom(prevTubbyCatOwner, alice.address, id)
+      tubbyCats = tubbyCats.connect(alice)
+      const ownerBefore = await tubbyCats.ownerOf(id)
+      expect(ownerBefore).to.eq(alice.address)
+
+      // Signature is invalid
+      advancedOrder.signature = '0xdeadbeef'
+
+      const calldata = seaportInterface.encodeFunctionData('fulfillAdvancedOrder', [
```

Comment:
In addition to advancedOrder I'd also do calldata setup by advanced order since they'r relevant to each other instead of having tubbycat owner assignment in between which is a separate. neat organized tests are a life saver to maintain.